src/msca/metrics/plan.md

```diff
@@ -0,0 +1,308 @@
+    ----------
+    data
+        DataFrame containing all required columns (obs, pred, weights, score).
+    """
+```
+
+```python
+def get_performance(
+    data: pd.DataFrame,
+    skill: str = "skill",
+    avg_by: list[str],
+    desc: bool = False
+    other_cols : list[str] | None = None,
+) -> pd.DataFrame:
+    """Get performance summary by averaging skill scores across specified groups.
+    
+    Parameters
+    ----------
+    data
+        DataFrame containing skill column and grouping columns.
+    skill
+        Column name containing skill scores. Defaults to "skill".
+    avg_by
+        List of column names to group by and average skill scores.
+        Duplicates will be dropped based on these columns.
+    desc
+        If False (default), returns ascending skill scores (worst performers first).
+        If True, returns descending skill scores (best performers first).
+    other_cols
+        List of other columns that the user wants returned in the dataframe for easier diagnosis (e.g location_name)
+        
+    Returns
+    -------
+    pd.DataFrame
+        Filtered copy of input data with avg_by columns, other_cols (if specified), 
+        averaged skill scores, and skill_rank column. Sorted by skill score.
+        Contains only the relevant columns so it can be saved separately 
+        (e.g., as skill_rank.csv) or merged back with original data.
+        Worst performing groups appear first when desc=False.
+    """
+```
+
+```python
+def get_model_objective(
+    data: pd.DataFrame, 
+    xmodel: XModel
+) -> float:
+    """Get model objective score.
+    """
+```
+
+## TODOs
+
+- TODO: Update so that get_weighted_mean can be used instead of Xmodel
+- TODO: Have differences between the levels of grouby for instance - want to have the reference score be weighted mean average across age-sex and the prediction score will be by the same but the overall skill will be by sex location
+
+## Usage Examples
+
+```python
+# Step 1: Create reference score using weighted mean
+reference_scores = get_weighted_mean(
+    data=data_observations,
+    val="obs", 
+    weights="weights", 
+    by=["age_group_id", "sex_id"]
+    # Uses default name="wt_mean"
+)
+
+# Step 2: Get model scores by sex and age using pred_kreg
+onemod_score = get_model_scores(
+    metric="mean_absolute_error",
+    data=data_modeling,  # Contains obs, pred_kreg, weights columns
+    groupby=["age_group_id", "sex_id"]
+    obs="obs",
+    pred="pred_kreg",
+    weights="weights"
+)
+
+# Step 3: Calculate skill scores comparing reference to onemod_score
+onemod_skill = get_skill_scores(
+    metric="mean_absolute_error",
+    data=data_modeling,
+    reference=reference_scores,
+    groupby=["age_group_id", "sex_id"],
+    obs="obs",
+    pred="pred_kreg", 
+    weights="weights",
+    score="wt_mean"  # Reference column from get_weighted_mean
+)
+
+# Step 4: Analyze performance to find worst performing groups
+worst_performers = get_performance(
```

Comment:
`get_performanace` function might not be in the scope of this design. Let's stick with the error metrics and the skill metric.